lib/modding/CModHandler.cpp

```diff
@@ -130,9 +137,11 @@ std::vector <TModID> CModHandler::validateAndSortDependencies(std::vector <TModI
 		if(!resolvedOnCurrentTreeLevel.empty())
 		{
 			resolvedModIDs.insert(resolvedOnCurrentTreeLevel.begin(), resolvedOnCurrentTreeLevel.end());
-					continue;
+			for(auto it = resolvedOnCurrentTreeLevel.begin(); it != resolvedOnCurrentTreeLevel.end(); it++)
+				notResolvedModIDs.erase(*it);
```

Comment:
```suggestion
			for(const auto & it : resolvedOnCurrentTreeLevel)
				notResolvedModIDs.erase(it);
```

Reply:
it is the Iterator of resolvedOnCurrentTreeLevel，is it safe in notResolvedModIDs.erase(it);？I dont know. So I follow your suggestion.